### DISCUSSION TITLE
Is StatefulWidget really needed?

### DISCUSSION THREAD

**Discussion by lemunozm** [fdehau/tui-rs#432] [General]
Hi! I was playing with `StatelessWidget` and `StatefulWidget` to choose the best approach for my target. 

But a question comes to my mind: is the `StatefulWidget` really necessary? There is something that I can do with it that I can not with a simple `StatelessWidget` with a mutable reference to the state?

I understand the concept behind this, but I think that inside of a paradigm where a widget is consumed after rendering loses its benefits.

For example, I can express the `StatefulWidget` behaviour with a `StatelessWidget` in the following way:

```rust
let mut state = ListState::default();

// With StatefulWidget (current behavior):
let list = List::new(items);
frame.render_statefull_windget(list, area, &mut state); // list has been consumed here

// With StatelesWidget: now List is a StatelessWidget with a new parameter &mut ListState
let list = List::new(items, &mut state);
frame.render_stateless_windget(list, area); // list has been consumed here
```
There is some real difference between these two ways?
Is there something that I am not seeing?
Thank you so much!!

**Comment by kvark** [comment]
I was just about filing the same issue, glad to discover this!
Yet, ~more than 2 years~ a year and a half without an answer, this is a bit worrying.

**Comment by kaixinbaba** [comment]
I have the same question that I can to save state through the fields in the structure of stateless in order to achieve the same purpose
